src/test/java/org/codehaus/plexus/util/xml/pull/MXParserTest.java

```diff
@@ -840,4 +846,56 @@ public void testXMLDeclVersionEncodingStandaloneNoSpace()
         }
     }
 
+    /**
+     * Issue 163: https://github.com/codehaus-plexus/plexus-utils/issues/163
+     *
+     * @throws IOException if IO error.
+     *
+     * @since 3.4.1
+     */
+    @Test
+    public void testEncodingISO_8859_1setInputReader()
+        throws IOException
+    {
+        try ( Reader reader =
+            ReaderFactory.newXmlReader( new File( "src/test/resources/xml", "test-encoding-ISO-8859-1.xml" ) ) )
+        {
+            MXParser parser = new MXParser();
+            parser.setInput( reader );
+            while ( parser.nextToken() != XmlPullParser.END_DOCUMENT )
+                ;
+            assertTrue( true );
+        }
+        catch ( XmlPullParserException e )
+        {
+            fail( "should not raise exception: " + e );
+        }
+    }
+
+    /**
+     * Issue 163: https://github.com/codehaus-plexus/plexus-utils/issues/163
+     *
+     * @throws IOException if IO error.
+     *
+     * @since 3.4.1
+     */
+    @Test
+    public void testEncodingISO_8859_1_setInputStream()
+        throws IOException
+    {
+        try ( InputStream input =
+            Files.newInputStream( Paths.get( "src/test/resources/xml", "test-encoding-ISO-8859-1.xml" ) ) )
+        {
+            MXParser parser = new MXParser();
+            parser.setInput( input, null );
+            while ( parser.nextToken() != XmlPullParser.END_DOCUMENT )
+                ;
```

Comment:
nit I would prefer nothing rather than this (maybe only me but looks weird this :) )